server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -96,9 +94,18 @@
   private final AtomicReference<ConcurrentHashMap<String, DruidDataSource>> dataSourcesRef;
   private final SQLMetadataConnector connector;
 
-  private volatile ListeningScheduledExecutorService exec = null;
-  private volatile ListenableFuture<?> future = null;
-  private volatile boolean started;
+  /** The number of times this SQLMetadataSegmentManager was started. */
+  private long startCount = 0;
+  /**
+   * Equal to the current {@link #startCount} value, if the SQLMetadataSegmentManager is currently started; -1 if
+   * currently stopped.
+   *
+   * This field is used to implement a simple stamp mechanism instead of just a boolean "started" flag to prevent
+   * the theoretical situation of two tasks scheduled in {@link #start()} calling {@link #poll()} concurrently, if
+   * the sequence of {@link #start()} - {@link #stop()} - {@link #start()} actions occurs quickly.
+   */
```

Comment:
Please cross-reference those things in both classes. Like I changed one class and didn't know that there is another one that demands a similar change. You knew that, but this knowledge should be in code for any code developer.

Reply:
Sounds a good convention. I also haven't noticed until I check the changes in https://github.com/apache/incubator-druid/pull/5554 which introduced this bug. 

Added a comment.